admin-perspective/olm/install-forcenamespace/install-forcenamespace.md

```diff
@@ -0,0 +1,16 @@
+# Operator-defined Namespace
+
+Operator bundles can define what namespace they should be installed to, so the user should be prevented from changing the target namespace.
+
+## When namespace exists
+
+![Operator install force namespace](img/1-1-forcenamespace.png)
```

Comment:
Should "A specific namespace on the cluster" actually be the item that's selected and then it will specify that openshift-logging has been chosen and can't be changed by the user?

Reply:
That was what I was expecting as well when first taking a look at this design, and I believe was how Riley had first drawn it. I found this comment where it was changed: https://docs.google.com/document/d/1hQR4QFYa05YHIFBFfZoVFB46ubtFjr-KE48fXuENjJY/edit?disco=AAAADeOK2Yc

It seems the the primary (only?) use case is that the operator will be installed cluster-wide, it just happens to live on a different namespace than openshift-operators where global operators usually are.

Is that still accurate @dmesser @tlwu2013 ?

Reply:
So basically the forced-namespace is a way how to specify the namespace in case of the "All namespaces on the cluster (default)" option, so the picked operator will be installed into namespace defined with the `operators.operatorframework.io.targetNamespace=<targetNamespace>` annotation  on the package manifest instead of the `openshift-operators` namespace. 
Right ?

Reply:
Is the assumption that whatever targetNamespace is specified will be available cluster-wide? Is there any scenario where the operator would be installed to a namespace where it is only available in that namespace? The idea of an operator being installed to a namespace other than openshift-operators and being available cluster-wide is still confusing to me. @bparees

Reply:
1) this should not be tied specifically to the "available to all namespaces on the cluster" path.  If an operator wants to recommend (please stop saying force, it's not a force) a namespace for itself, that recommendation should be used whether we are installing the operator "globally" or for a specific namespace (it just becomes the default specific namespace to use).

2) the reason this is needed is because we cannot scrape telemetry data out of "openshift-operators" because it contains community operators that could report their own metrics we don't want in telemeter.  But we do have some global operators (elastic search being the canonical example) that do need to report telemeter data.  Thus they need to be installed into a namespace that can be scraped (ie not openshift-operators) and still be available cluster-wide.

Reply:
That is helpful. So the user should still be able to choose All namespaces (global) or specific namespace to install the operator to. If global, default to targetNamespace with option to switch back to openshift-operators. If specific, use the existing control defaulted to targetNamespace, with the option to switch to any other namespace.

For my own understanding, and perhaps to guide the UX, is there some other way an operator is known to be global, besides it being installed to openshift-operators? Up until now my understanding was operators in openshift-operators were global, operators in any other namespace were specific to that namespace.

Reply:
> For my own understanding, and perhaps to guide the UX, is there some other way an operator is known to be global, besides it being installed to openshift-operators? Up until now my understanding was operators in openshift-operators were global, operators in any other namespace were specific to that namespace.


it's based on the operator-group and/or subscription.  i'm not positive on the precise api.  @ecordell can tell us.

Reply:
> is there some other way an operator is known to be global, besides it being installed to openshift-operators?

If one of the operator's `OperatorGroup` has name `global-operator` then the `openshift-operators` namespace is used